futures-await-macro/src/lib.rs

```diff
@@ -22,7 +22,15 @@ macro_rules! await {
                     break ::futures::__rt::Err(e)
                 }
             }
-            yield
+            yield ::futures::Async::NotReady
         }
     })
 }
+
+#[macro_export]
+macro_rules! stream_yield {
+    ($e:expr) => ({
+        let e = $e;
+        yield ::futures::Async::Ready(e)
```

Comment:
Could this perhaps just be `yield e`? I think it'd be fine to use two different conversions internally for the two attributes

Reply:
I don't understand what you mean by two different conversions here, could you explain a bit more?

Reply:
Ah sure yeah, basically I just mean that yielding in `#[async]` should be just a literal `yield`, and yielding in `#[async_stream]` would be `yield e`

Reply:
So instead of having the `stream_yield!` macro, just do a rewrite from `yield e;` to `yield ::futures::Async::Ready(e);` as part of the `#[async_stream]` macro? That does have the downside of meaning you can't write macros that yield values, but I assume those would be rare enough to not matter. I'll try and get this and the attribute parsing done tomorrow.

Reply:
Oh no I was thinking we'd keep the `stream_yield!` macro, but it would be implemented with `yield e`

Reply:
Oh, how would that fit with `await!` and `#[async] for` yielding `Async::NotReady` then, just have the user do something like `stream_yield!(Async::Ready(e))`? I thought part of the purpose of async/await syntax was to allow composing futures/streams together without having to work with some of the more verbose underlying types like `Async`.

Reply:
Ah I see, that makes sense! In that case let's leave as-is.

Reply:
Depending on how much magic (tm) is acceptable, `#[async]` and `#[async_stream]` could themselves expand `await!` in different ways that make sense for each of them?

Reply:
Nah I'm fine using the strategy here, seems fine to me!